src/main/scala/is/hail/rvd/RVD.scala

```diff
@@ -137,7 +151,58 @@ trait RVD {
 
   def crdd: ContextRDD[RVDContext, RegionValue]
 
-  def rdd: RDD[RegionValue]
+  private[rvd] def stabilize(
+    unstable: ContextRDD[RVDContext, RegionValue]
+  ): ContextRDD[RVDContext, RegionValue] = {
+    val localRowType = rowType
+    unstable.map { rv =>
+      val region = Region()
+      val rvb = new RegionValueBuilder(region)
+      rvb.start(localRowType)
+      rvb.addRegionValue(localRowType, rv)
+      RegionValue(region, rvb.end())
+    }
+  }
+
+  private[rvd] def stably(
+    f: ContextRDD[RVDContext, RegionValue] => ContextRDD[RVDContext, RegionValue]
+  ): ContextRDD[RVDContext, RegionValue] = f(stabilize(crdd))
+
+  private[rvd] def stably(
+    unstable: ContextRDD[RVDContext, RegionValue],
+    f: ContextRDD[RVDContext, RegionValue] => ContextRDD[RVDContext, RegionValue]
+  ): ContextRDD[RVDContext, RegionValue] = f(stabilize(unstable))
+
+  private[rvd] def crddBoundary: ContextRDD[RVDContext, RegionValue] =
+    crdd.cmapPartitionsAndContext { (consumerCtx, part) =>
+      val producerCtx = consumerCtx.freshContext
+      val it = part.flatMap(_ (producerCtx))
+      new Iterator[RegionValue]() {
+        private[this] var cleared: Boolean = false
+
+        def hasNext = {
+          if (!cleared) {
+            cleared = true
+            producerCtx.region.clear()
+          }
+          it.hasNext
+        }
+
+        def next = {
+          if (!cleared) {
+            producerCtx.region.clear()
+          }
+          cleared = false
+          it.next
+        }
+      }
+    }
+
+  // SOMEWHAT UNSAFE: the values produced will not be in the consumer's region,
+  // you must be careful to copy the value into the appropriate region
+  def boundary: RVD
+
+  def rdd: RDD[RegionValue] = stabilize(crdd).run
```

Comment:
This needs to go, right?  We can't have a un-contextualized RDD[RegionValue], can we?

Reply:
Yeah, that's right.

This will have to produce `RDD[Array[Byte]]` or `RDD[Annotation]`. I'll take a look at the uses and determine what the right choice is (or maybe we need both?).